src/json/serialization.cr

```diff
@@ -216,26 +216,24 @@ module JSON
           case key
           {% for name, value in properties %}
             when {{value[:key]}}
-              %found{name} = true
               begin
-                %var{name} =
-                  {% if value[:nilable] || value[:has_default] %} pull.read_null_or { {% end %}
-
-                  {% if value[:root] %}
-                    pull.on_key!({{value[:root]}}) do
-                  {% end %}
-
-                  {% if value[:converter] %}
-                    {{value[:converter]}}.from_json(pull)
-                  {% else %}
-                    ::Union({{value[:type]}}).new(pull)
-                  {% end %}
+                {% if value[:has_default] || value[:nilable] %} pull.read_null_or do {% else %} begin {% end %}
```

Comment:
Is `begin` actually necessary here or is it just the counterpart for an unconditional `end` later?

Reply:
The existing code guarded both the `{` and the `}`, but I think the code indents better this way.